hibernate-core/src/main/java/org/hibernate/internal/SessionImpl.java

```diff
@@ -3531,8 +3531,14 @@ public void remove(Object entity) {
 			throw exceptionConverter.convert( new IllegalArgumentException( e.getMessage(), e ) );
 		}
 		catch ( JDBCException e ) {
-			if ( accessTransaction().getRollbackOnly() ) {
-				// assume this is the similar to the WildFly / IronJacamar "feature" described under HHH-12472
+			if ( accessTransaction().isActive() && accessTransaction().getRollbackOnly() ) {
+				// Assume this is the similar to the WildFly / IronJacamar "feature" described under HHH-12472.
+				// Just log the exception and return null.
+				if ( log.isDebugEnabled() ) {
+					log.debug( "JDBCException was thrown for a transaction marked for rollback; " +
+									   "this is probably due to IronJacamar failing fast after " +
```

Comment:
Would it be okay to generalize the logged message to not mention IronJacamar, in case IronJacamar is not in use, perhaps something like:

```
log.debug( "JDBCException was thrown for a transaction marked for rollback; " +
									   "this is probably due to an operation failing fast due to " +
									   "transaction marked for rollback.", e );

```

Reply:
@scottmarlow , yes, good idea; I'll do that.

I see there's a checkstyle failure, so I'll fix that as well.